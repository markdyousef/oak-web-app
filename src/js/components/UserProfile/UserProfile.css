:root {
    --main-black-color: #000;
    --main-white-color: #fefeff;
    --main-orange-color: #ff9200;
    --main-red-color: #8b0000;
    --main-green-color: #005d6c;
}

.container {
    width: 100%;
}

.container header {
    display: flex;
    flex-direction: column;
    background-color: #fff;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
}

.container h1 {
    font-weight: bold;
    align-self: flex-start;
    font-size: 16px;
    margin: 20px 0;
}

.container h3 {
    align-self: flex-start;
    margin: 10px 0;
}

.profile {
    margin-top: 30px;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.profile h1 {
    margin-top: 20px;
    font-size: 20px;
}

.avatar {
    height: 150px;
    width: 150px;
    border-radius: 100px;
}

.compare {
    padding: 20px;
}

.stats {
    margin-top: 30px;
    margin-bottom: 30px;
}

.users {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
}

.users span {
    border: 0 !important;
    padding: 0 !important;
    height: 30px;
    width: 30px;
}
