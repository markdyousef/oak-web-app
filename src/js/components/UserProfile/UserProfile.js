import React, { Component, PropTypes } from 'react';

import css from './UserProfile.css';
import Avatar from '../shared/Avatar';
import Insights from '../Insights';

class UserProfile extends Component {
    static propTypes = {
        userId: PropTypes.string.isRequired,
        user: PropTypes.shape({
            profile: PropTypes.object
        }),
        getInsight: PropTypes.func.isRequired
    }
    static defaultProps = {
        user: {
            profile: {}
        }
    }
    constructor() {
        super();
        this.state = {}
    }
    componentWillMount() {
        const { getInsight, userId } = this.props;
        getInsight(userId);
    }
    render() {
        const { user, insights } = this.props;
        return (
            <div className={css.container}>
                <header>
                    <div className={css.profile}>
                        <div className={css.avatar}>
                            <Avatar img={user.profile.image_original} />
                        </div>
                        <Insights insights={insights} />
                    </div>
                </header>
            </div>
        );
    }
}

export default UserProfile;
