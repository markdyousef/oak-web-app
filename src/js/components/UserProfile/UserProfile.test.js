// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import UserProfile from './UserProfile';

test('renders UserProfile', () => {
    const props = {
        userId: 'userId',
        getInsight: () => {},
        getUser: () => {}
    };
    const component = renderer.create(
        <UserProfile {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
