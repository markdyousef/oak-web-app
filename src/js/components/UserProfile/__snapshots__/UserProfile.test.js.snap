exports[`test renders UserProfile 1`] = `
<div
  className={undefined}>
  <header>
    <div
      className={undefined}>
      <div
        className={undefined}>
        <div
          className={undefined}>
          <img
            alt="profile"
            src={null} />
        </div>
      </div>
      <div
        className={undefined} />
    </div>
  </header>
</div>
`;
